### ISSUE TITLE
[Bug]: `aws_eks_cluster` opting into EKS Auto Mode with built in node pools is currently requiring the cluster to be replaced

### ISSUE THREAD

**Issue by bryantbiggs** [hashicorp/terraform-provider-aws#40411]
### Terraform Core Version

Any

### AWS Provider Version

5.79.0

### Affected Resource(s)

* aws_eks_cluster

### Expected Behavior

Users should be able to opt into EKS Auto Mode with existing clusters without requiring the cluster to be recreated. This is possible without using the built in nodepools, but currently not possible if users utilize one of the built in nodepools (`system`, `general-purpose`) 

The set of behaviors are:
1. Enabling `compute_config.enabled = true` and setting a `compute_config.node_role_arn` is a valid in-place change
2. Disabling `compute_config.enabled = false` even with a `compute_config.node_role_arn` specified is a valid in-place change
3. With `compute_config.enabled = true`, changing `compute_config.node_role_arn` from one ARN value to another is not a valid in-place change; this change can only be made by re-placing the cluster

More context: these behaviors are defined due to the use of the IAM role by the instances created by EKS Auto Mode. Removing the ARN when disabling `compute_config` is valid because the instances that use the ARN will be destroyed/removed. Changing the ARN in-place would require changing the IAM role on *all instances that have been created by EKS Auto Mode 

### Actual Behavior

Opting into EKS Auto Mode with an existing cluster and specifying a `node_role_arn` is wanting to re-create the cluster since the `node_role_arn` value is changing (from undefined to defined) 

### Relevant Error/Panic Output Snippet

```shell
No panic or error
```

### Terraform Configuration Files

See steps to repro below - will open a draft PR that includes the updated acceptance test that demonstrates the issue

### Steps to Reproduce

1. Deploy a basic EKS cluster using this example https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_cluster#eks-cluster
2. After the cluster is deployed, add the EKS Auto Mode configuration shown in this example https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_cluster#eks-cluster-with-eks-auto-mode and run `terraform plan`
3. You will see that the cluster is slated to be replaced due to the change in the `compute_config.node_role_arn`

### Would you like to implement a fix?

Yes

**Comment by jatinmehrotra** [comment]
@bryantbiggs 

- I am trying to enable eks auto mode on existing cluster
- However atm `cluster_compute_config` only allows `general-purpose, system` as node pools value
- i need to create custom nodepool becuse as you mnentioned its not possible opt in Auto mode for existing cluster if we use  `general-purpose, system` as built in nodepools

**Comment by jatin-mehrotra-colorkrew** [comment]
@bryantbiggs 

https://github.com/hashicorp/terraform-provider-aws/issues/40411#issuecomment-2519631157

This issue is also true for custom node pools too even though aws docs support creating node pools https://docs.aws.amazon.com/eks/latest/userguide/create-node-pool.html but there is option to specify custom node pools too.

**Comment by magzim21** [comment]
This is still an issue
![CleanShot-2024-12-11-at-21 36 19@2x](https://github.com/user-attachments/assets/2884101a-d5f6-4b44-99fc-de819df88a70)
I upgraded aws provider to `5.80` , using the latest "terraform-aws-modules/eks/aws" module.
An existing cluster is to be replaced.


```tf
module "eks" {
  source  = "terraform-aws-modules/eks/aws"
  version = "20.31.1"
...
  cluster_compute_config = {
    enabled    = true
    node_pools = ["system","general-purpose"]
  }

```

**Comment by jatin-mehrotra-colorkrew** [comment]
@magzim21

No you are not right because it has been fixed and will be shipped in v5.80.1 and not 5.80.0 so please wait for it to be released

**Comment by magzim21** [comment]
> @magzim21
> 
> 
> 
> No you are not right because it has been fixed and will be shipped in v5.80.1 and not 5.80.0 so please wait for it to be released

I assumed that merging is releasing.

**Comment by jatin-mehrotra-colorkrew** [comment]
@magzim21 

This should help you to sleep : https://dev.to/aws-builders/eks-auto-mode-unlocked-for-existing-clusters-with-terraform-545i
